### DISCUSSION TITLE
Save or backup huge pdf embeddings

### DISCUSSION THREAD

**Discussion by sanketpatel91** [imartinez/privateGPT#1726] [Q&A]
Hello

Is there any way to save or backup  embeddings or data from already embedded huge pdf file and re use/ load again when required?

**Comment by JTMarsh556** [comment]
Yea, I was wondering if there is a way to push ingested files to a long term memory on the hard drive that can be queried and loaded into vRAM in a pre-ingested state.

**Comment by BradHutchings** [comment]
If you restart your computer and the privateGPT system, it should behave as if the listed files are ingested. I just tested this explicitly with a couple of short files describing both of my dogs, and having the chat make up a cute story about them. It worked as I expected. I've also observed that ingested files are resilient across restarts previously, but wanted to make double sure for you. I'll make a video about this soon! My system is in a Hyper-V virtual machine and doesn't have a GPU, but I'd expect a system with a GPU would reload these at startup.

**Reply by JTMarsh556** [reply]
They do reload.  My long term goal isn't really possible with vram alone so I'm hoping there is a way to ingest with long term storage on the drive.  Then query the ingested chunks on the drive, load them in the vram and use them.  

Dynamically loading and unloading chunks based on relevancy.

**Comment by tlinnet** [comment]
The vector is by default Qdrant as filebased. So persistent, unless if you use Docker without volume.
